test/test-kexdh.js

```diff
@@ -0,0 +1,79 @@
+var SSH2Stream = require('../lib/ssh');
+var MESSAGE = require('../lib/constants').MESSAGE;
+
+var basename = require('path').basename,
+    inspect = require('util').inspect,
+    assert = require('assert'),
+    path = require('path'),
+    fs = require('fs');
+
+var t = -1,
+    group = path.basename(__filename, '.js') + '/',
+    SERVER_KEY = fs.readFileSync(__dirname + '/fixtures/ssh_host_rsa_key');
+
+function multi_kexdh_init() {
+  function run() {
+    var what = 'MULTI_KEXDH_INIT: ';
```

Comment:
Minor nit: This could probably be simplified more to just be something like: `'Multi-KEXDH init'`